src/xcode/ENA/ENA/Source/Scenes/QRScanner/_tests_/QRScannerViewModelTests.swift

```diff
@@ -0,0 +1,286 @@
+					XCTAssertEqual(person.name?.standardizedName, "MITCHELL ROBERT", "Expected correct person standardizedName")
+					XCTAssertEqual(certificate.type, .vaccination, "Expected correct person standardizedName")
+					onSuccessExpectation.fulfill()
+				default:
+					XCTFail("Expected a successful scan of Vaccination Certificate")
+				}
+			case .failure:
+				XCTFail("Expected a successful scan of Vaccination Certificate")
+			}
+		}
+		
+		let validCertificateURL = "HC1:6BFOXN*TS0BI$ZD8UHRTHZDE+VJG$L20I7*S1RO4.S-OPL-I0$E7XHQI5UTE8X6F/8X*G-O9UVPQRHIY1VS1NQ1 WUQRELS4HBT5IV6:IR4DN8TLS4J1TOX0O05L%2NS4KCTIYC1MFE67EHNTGV6ALD-I:X0YO74$0NS4.$S6ZC0JBX63*E3IFTIJ3N:IN1MPF5RBQ746B46O1N646RM93O5RF6$T61R6B46646O$9KZ56DE/.QC$Q3J62:6PV66RU%TE6UG+ZE V1+GOT*OBR7UX4NC6O67U96L*KDYPKCOJADBIHOHK1JAF.7H:SZJJ%U4:7P CF7:4OHT-3TB6JS1J6:IZW4I:A6J3ARN5TTHR98D39RM3P0PLN5:FX K.8N IJ67M.EN37J+ULD7J.HK AE29OJDNCZG$XQ* 646L+7MEKRJTEYYDYAU:0P7UUR9J6BM%BA0FE-VUYN73TTMF0UPM-73 OJS836VN++UTLISVIXE240HURI4UFPHLE*R%9N%2RB40.CKW3"
+		let metaDataObject = FakeMetadataMachineReadableCodeObject(stringValue: validCertificateURL)
+		viewModel.activateScanning()
+		viewModel.didScan(metadataObjects: [metaDataObject])
+		waitForExpectations(timeout: .short)
+	}
+	
+	func test_ifInValid_QRCode_Scanned_then_parsing_fails() {
+		let store = MockTestStore()
+		let client = ClientMock()
+		let appConfigurationProvider = CachedAppConfigurationMock()
+		let dscListProvider = MockDSCListProvider()
+		let dccSignatureVerifier = DCCSignatureVerifyingStub()
+		let boosterNotificationsService = BoosterNotificationsService(
+			rulesDownloadService: RulesDownloadService(store: store, client: client)
+		)
+		let healthCertificateService = HealthCertificateService(
+			store: store,
+			dccSignatureVerifier: dccSignatureVerifier,
+			dscListProvider: dscListProvider,
+			client: client,
+			appConfiguration: appConfigurationProvider,
+			boosterNotificationsService: boosterNotificationsService
+		)
+		let verificationService = QRCodeVerificationHelper()
+		let onFailureExpectation = expectation(description: "onFailure called")
+		onFailureExpectation.expectedFulfillmentCount = 1
+
+		let viewModel = TestQRScannerViewModel(
+			healthCertificateService: healthCertificateService,
+			verificationHelper: verificationService,
+			appConfiguration: appConfigurationProvider) { result in
+			switch result {
+			case .success(let result):
+				switch result {
+				case .certificate(let person, let certificate):
+					XCTAssertEqual(person.dateOfBirth, "1981-08-30", "Expected correct person dateOfBirth")
+					XCTAssertEqual(person.name?.standardizedName, "MITCHELL ROBERT", "Expected correct person standardizedName")
+					XCTAssertEqual(certificate.type, .vaccination, "Expected correct person standardizedName")
+					onFailureExpectation.fulfill()
+				default:
+					XCTFail("Expected a successful scan of Vaccination Certificate")
+				}
```

Comment:
```suggestion
				XCTFail("Expected scan to fail")
```

This should not succeed, right?

Reply:
yeah correct